src/payment/checkout/payment-form/flex-microform/FlexMicroform.jsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import CreditCardNumberField from './CreditCardNumberField';
+import CreditCardVerificationNumberField from './CreditCardVerificationNumberField';
+import { DEFAULT_STATUS, STATUS_READY } from './constants';
+
+const POLLING_TIME_MS = 100;
+
+class FlexMicroform extends React.Component {
+  constructor(props) {
+    super(props);
+
+    window.microform = null;
+    this.state = {
+      microformStatus: DEFAULT_STATUS,
+    };
+    this.timer = null;
+  }
+
+  componentDidMount() {
+    this.initialize();
+  }
+
+  componentDidUpdate() {
+    this.initialize();
+  }
+
+  componentWillUnmount() {
+    if (this.timer) {
+      clearTimeout(this.timer);
+      this.timer = null;
+    }
+  }
+
+  initialize = () => {
+    if (window.microform !== null) {
+      return;
+    }
+    if (!this.props.captureKeyId) {
+      return;
+    }
+    if (this.timer) {
+      clearTimeout(this.timer);
+      this.timer = null;
+    }
+    if (typeof window.Flex === 'undefined') {
+      this.timer = setTimeout(this.initialize, POLLING_TIME_MS);
+      return;
+    }
+    window.microform = new window.Flex(this.props.captureKeyId).microform({
```

Comment:
This "captureKeyId" is a JWT. It is valid for 15 minutes. We need to handle the case when it expires, presumably we need to make an AJAX request to ecommerce to refresh the token and then re-initialize `window.microform`. I suspect this will wipe out whatever input the user had entered, but that's something we need to figure out.

Reply:
They symptom of an expired token is that `microform.createToken()` will return an error.